pkg/tags/tag.go

```diff
@@ -62,18 +65,22 @@ type Tag struct {
 	StartedAt time.Time     // tag started to calculate ETA
 
 	// end-to-end tag tracing
-	ctx      context.Context  // tracing context
-	span     opentracing.Span // tracing root span
-	spanOnce sync.Once        // make sure we close root span only once
+	ctx        context.Context     // tracing context
+	span       opentracing.Span    // tracing root span
+	spanOnce   sync.Once           // make sure we close root span only once
+	stateStore storage.StateStorer // to persist the tag
```

Comment:
We are now passing the database object through the context, which brings its own problems and is discouraged to do so. I think that tags would benefit from a different architecture where only the tag id would be passed in the context, and a tag service would be responsible for getting, incrementing and storing counters.

Reply:
I agree. But this is out of the scope of this PR.